File: hw2/hw2_resources/pegasos_gaussian_test.m
function pegasos_test

% load data from csv files
data = importdata(strcat('data/data3_train.csv'));
global X
X = data(:,1:2);
Y = data(:,3);
[n, d] = size(X);


disp('======Gaussian Kernel SVM======');
% Carry out training.
global alpha
global gamma
global K
epochs = 1000;
lambda = .02;
gamma = 2^2;

K = zeros(n,n);
%%% TODO: Compute the kernel matrix %%%

%%% TODO: Implement train_gaussianSVM %%%
train_gaussianSVM(X, Y, lambda, epochs);


% Define the predict_gaussianSVM(x) function, which uses trained parameters, alpha
%%% TODO:  define predict_gaussianSVM(x) %%%


hold on;

% plot training results
plotDecisionBoundary(X, Y, @predict_gaussianSVM, [-1,0,1], 'Gaussian Kernel SVM');

File: 6.867/hw2/hw2_resources/pegasos_gaussian_test.py
from numpy import *
from plotBoundary import *
import pylab as pl
# import your LR training code

# load data from csv files
train = loadtxt('data/data3_train.csv')
X = train[:,0:2]
Y = train[:,2:3]

# Carry out training.
epochs = 1000;
lmbda = .02;
gamma = 2e-2;

K = zeros((n,n));
### TODO: Compute the kernel matrix ###

### TODO: Implement train_gaussianSVM ###
alpha = train_gaussianSVM(X, Y, K, lmbda, epochs);


# Define the predict_gaussianSVM(x) function, which uses trained parameters, alpha
### TODO:  define predict_gaussianSVM(x) ###

# plot training results
plotDecisionBoundary(X, Y, predict_gaussianSVM, [-1,0,1], title = 'Gaussian Kernel SVM')
pl.show()
